test/api/v3/integration/groups/POST-groups_groupId_leave.js

```diff
@@ -0,0 +1,147 @@
+
+          await leader.post(`/tasks/challenge/${challenge._id}`, {
+            text: 'test habit',
+            type: 'habit',
+          });
+          await sleep(2);
+        });
+
+        it('removes all challenge tasks when keep parameter is set to remove', async () => {
+          await leader.post(`/groups/${groupToLeave._id}/leave?keep=remove-all`);
+
+          let userWithoutChallenges = await leader.get('/user');
+
+          expect(userWithoutChallenges.challenges.indexOf(challenge._id)).to.equal(-1);
+          expect(userWithoutChallenges.tasksOrder.habits.length).to.equal(0);
+        });
+
+        it('keeps all challenge tasks when keep parameter is not set', async () => {
+          await leader.post(`/groups/${groupToLeave._id}/leave`);
+
+          let userWithChallenges = await leader.get('/user');
+
+          expect(userWithChallenges.challenges.indexOf(challenge._id)).to.equal(-1);
+          expect(userWithChallenges.tasksOrder.habits.length).to.equal(1);
+        });
+      });
+
+      xit('prevents quest leader from leaving a groupToLeave', async () => {});
+      xit('prevents a user from leaving during an active quest', async () => {});
+    });
+  });
+
+  context('Leaving a group as the last member', () => {
+    it('removes a group and invitations when the last member leaves a private guild', async () => {
+      let { group, groupLeader, invitees } = await createAndPopulateGroup({
+        groupDetails: {
+          name: 'Test Private Guild',
+          type: 'guild',
+        },
+        invites: 1,
+      });
+
+      await groupLeader.post(`/groups/${group._id}/leave`);
+
+      let groups = await groupLeader.get('/groups?type=party,privateGuilds,publicGuilds,tavern');
+      let userWithoutInvitation = await invitees[0].get('/user');
+
+      expect(_.findIndex(groups, {_id: group._id})).to.equal(-1);
+      expect(userWithoutInvitation.invitations.guilds).to.be.empty;
+      await expect(checkExistence('groups', group._id)).to.eventually.equal(false);
+    });
+
+    it('removes invitations but not the group when the last member leaves a public guild', async () => {
+      let { group, groupLeader, invitees } = await createAndPopulateGroup({
+        groupDetails: {
+          name: 'Test Public Guild',
+          type: 'guild',
+          privacy: 'public',
+        },
+        invites: 1,
+      });
+
+      await groupLeader.post(`/groups/${group._id}/leave`);
+
+      let groups = await groupLeader.get('/groups?type=party,privateGuilds,publicGuilds,tavern');
+      let userWithoutInvitation = await invitees[0].get('/user');
+
+      expect(_.findIndex(groups, {_id: group._id})).to.be.above(-1);
+      expect(userWithoutInvitation.invitations.guilds).to.be.empty;
+      await expect(checkExistence('groups', group._id)).to.eventually.equal(true);
+    });
+
+    it('removes a group and invitations when the last member leaves a party', async () => {
```

Comment:
Again, these should be separate tests. One for the group removal, and one for the invitation removal.